### PR TITLE
feat: use dhis config provider to setup app store

### PR DESCRIPTION
**Pull request by varl**
DHIS2-7257

Move APP_STORE_URL and APP_STORE_API_URL from runtime configurable
system settings to boot-time configured through the `dhis.conf` file.

This still uses the current app store (v1) as the default, and allows the
user to configure the new app store (v2) through `dhis.conf`.

```
appstore.base.url=https://default/url/to/appstore
appstore.api.url=https://default/url/to/appstore/api
appstore.api.version=v1
```

This also cleans up the very old JSON based appstore as it was unused in
the codebase, and renames the "appstore2" package to "appstore".

### CODE REVIEW

**Comment by varl** [comment]
Since the old App Store API uses a single unversioned endpoint, it was hardcoded to `https://play.dhis2.org/appstore/api/apps` which returns all the applications on the app store.

The new App Store API is versioned and supports additional endpoints than the singular one from before, so the configuration options now look like:

```
appstore.base.url = http://dhis2-app-store-prod.eu-west-1.elasticbeanstalk.com/
appstore.api.url = http://dhis2-app-store-prod.eu-west-1.elasticbeanstalk.com/api
appstore.api.version = v1
```

The defaults in the `ConfigurationKey` class are designed to match the old app store for now.

The idea is to backport the configuration style to the supported versions, and then switch the default in `master` when we feel comfortable using the new app store backend.

File: dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/appstore/DefaultAppStoreService.java [hunk 1]
```diff
@@ -53,34 +55,47 @@
...
 
-        WebApp[] apps = restTemplate.getForObject( appStoreApiUrl, WebApp[].class );
+        if ( "" == appStoreApiVersion )
+        {
+            allAppsUrl = appStoreApiUrl + "/apps";
+        }
+        else
+        {
+            allAppsUrl = appStoreApiUrl + "/" + appStoreApiVersion + "/apps";
+        }
```
Comment (varl) [inline dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/appstore/DefaultAppStoreService.java]: This matches the hardcoded behaviour of today, but when the core starts talking to the app store api using new endpoints this logic should be abstracted out.

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/appmanager/DefaultAppManager.java [hunk 2]
```diff
@@ -74,18 +76,18 @@
 
     private final CacheProvider cacheProvider;
 
-    public DefaultAppManager( SystemSettingManager settingManager, CurrentUserService currentUserService,
-        LocalAppStorageService localAppStorageService, JCloudsAppStorageService jCloudsAppStorageService,
-        KeyJsonValueService keyJsonValueService, CacheProvider cacheProvider )
+    public DefaultAppManager(DhisConfigurationProvider dhisConfigurationProvider, CurrentUserService currentUserService,
```
Comment (stian-sandvold) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/appmanager/DefaultAppManager.java]:
```suggestion
    public DefaultAppManager( DhisConfigurationProvider dhisConfigurationProvider, CurrentUserService currentUserService,
```

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/appmanager/DefaultAppManager.java [hunk 3]
```diff
@@ -277,13 +279,7 @@ public boolean markAppToDelete( App app )
...
     public String getAppStoreUrl()
     {
-        return StringUtils.trimToNull( (String) settingManager.getSystemSetting( SettingKey.APP_STORE_URL ) );
-    }
-
-    @Override
-    public void setAppStoreUrl( String appStoreUrl )
-    {
-        settingManager.saveSystemSetting( SettingKey.APP_STORE_URL, appStoreUrl );
+        return StringUtils.trimToNull(dhisConfigurationProvider.getProperty( ConfigurationKey.APP_STORE_URL ) );
```
Comment (stian-sandvold) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/appmanager/DefaultAppManager.java]:
```suggestion
        return StringUtils.trimToNull( dhisConfigurationProvider.getProperty( ConfigurationKey.APP_STORE_URL ) );
```

File: dhis-2/dhis-services/dhis-service-setting/src/test/java/org/hisp/dhis/setting/SystemSettingManagerTest.java [hunk 4]
```diff
@@ -133,13 +129,12 @@ public void testGetSystemSettingsAsMap()
...
+            .of( SettingKey.APPLICATION_TITLE, SettingKey.APPLICATION_INTRO );
 
-        systemSettingManager.saveSystemSetting( APP_STORE_URL, "valueA" );
-        systemSettingManager.saveSystemSetting( APPLICATION_TITLE, "valueB" );
-        systemSettingManager.saveSystemSetting( APPLICATION_INTRO, "valueC" );
+        systemSettingManager.saveSystemSetting( APPLICATION_TITLE, "valueA" );
+        systemSettingManager.saveSystemSetting( APPLICATION_INTRO, "valueB" );
 
-        assertEquals( systemSettingManager.getSystemSettings( keys ).size(), 3);
+        assertEquals( systemSettingManager.getSystemSettings( keys ).size(), 2);
```
Comment (stian-sandvold) [inline dhis-2/dhis-services/dhis-service-setting/src/test/java/org/hisp/dhis/setting/SystemSettingManagerTest.java]:
```suggestion
        assertEquals( systemSettingManager.getSystemSettings( keys ).size(), 2 );
```

File: dhis-2/dhis-support/dhis-support-external/src/main/java/org/hisp/dhis/external/conf/ConfigurationKey.java [hunk 5]
```diff
@@ -99,7 +99,10 @@
     MONITORING_DBPOOL_ENABLED( "monitoring.dbpool.enabled", "off", false ),
     MONITORING_HIBERNATE_ENABLED( "monitoring.hibernate.enabled", "off", false ),
     MONITORING_UPTIME_ENABLED( "monitoring.uptime.enabled", "off", false ),
-    MONITORING_CPU_ENABLED( "monitoring.cpu.enabled", "off", false );
+    MONITORING_CPU_ENABLED( "monitoring.cpu.enabled", "off", false ),
+    APP_STORE_URL( "appstore.base.url", "https://play.dhis2.org/appstore", false),
```
Comment (stian-sandvold) [inline dhis-2/dhis-support/dhis-support-external/src/main/java/org/hisp/dhis/external/conf/ConfigurationKey.java]:
```suggestion
    APP_STORE_URL( "appstore.base.url", "https://play.dhis2.org/appstore", false ),
```

File: dhis-2/dhis-support/dhis-support-external/src/main/java/org/hisp/dhis/external/conf/ConfigurationKey.java [hunk 6]
```diff
@@ -99,7 +99,10 @@
...
+    APP_STORE_API_URL( "appstore.api.url", "https://play.dhis2.org/appstore/api", false),
```
Comment (stian-sandvold) [inline dhis-2/dhis-support/dhis-support-external/src/main/java/org/hisp/dhis/external/conf/ConfigurationKey.java]:
```suggestion
    APP_STORE_API_URL( "appstore.api.url", "https://play.dhis2.org/appstore/api", false ),
```

File: dhis-2/dhis-support/dhis-support-external/src/main/java/org/hisp/dhis/external/conf/ConfigurationKey.java [hunk 7]
```diff
@@ -99,7 +99,10 @@
...
+    APP_STORE_API_VERSION("appstore.api.version", "", false);
```
Comment (stian-sandvold) [inline dhis-2/dhis-support/dhis-support-external/src/main/java/org/hisp/dhis/external/conf/ConfigurationKey.java]:
```suggestion
    APP_STORE_API_VERSION( "appstore.api.version", "", false );
```

**Comment by varl** [comment]
> 
> 
> Looks good to me @varl ,
> 
> Do you want to await some feedback from @larshelge before merging, or should I just merge it after the codestyle issues are resolved?

Thanks for the review @stian-sandvold, I was pretty sure I set up the `DHIS2Formatter.xml` but looks like it didn't run everywhere. I've applied the code style violations.

The main thing in this PR I'm not entirely sure about is that I wiped these files: https://github.com/dhis2/dhis2-core/tree/master/dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/appstore

Here is my reasoning:

- I could not find any references to `org.hisp.dhis.appstore` in code
- The JSON that backs that implementation is very out of date and **none of the links in it work**: https://s3-eu-west-1.amazonaws.com/dhis2-appstore/appstore.json
- The implementation predates `org.hip.dhis.appstore2` which is used in all supported versions of DHIS2

I would be very happy to get this in ASAP since it blocks progression on the App Store.

**Comment by stian-sandvold** [comment]
> > Looks good to me @varl ,
> > Do you want to await some feedback from @larshelge before merging, or should I just merge it after the codestyle issues are resolved?
> 
> Thanks for the review @stian-sandvold, I was pretty sure I set up the `DHIS2Formatter.xml` but looks like it didn't run everywhere. I've applied the code style violations.
> 
> The main thing in this PR I'm not entirely sure about is that I wiped these files: https://github.com/dhis2/dhis2-core/tree/master/dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/appstore
> 
> Here is my reasoning:
> 
> * I could not find any references to `org.hisp.dhis.appstore` in code
> * The JSON that backs that implementation is very out of date and **none of the links in it work**: https://s3-eu-west-1.amazonaws.com/dhis2-appstore/appstore.json
> * The implementation predates `org.hip.dhis.appstore2` which is used in all supported versions of DHIS2
> 
> I would be very happy to get this in ASAP since it blocks progression on the App Store.

I'm not sure why, but there seems to be cases where the codestyle is not formatting properly. I might try and set off some time to review it.

I agree with your reasoning. It seems like some leftover files that we don't use anymore. It should be safe to delete it.

I'll merge this tonight when I get home, so travis can finish up

**Pull request merged**